.jenkinsci/helpers/hash.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+
+import hashlib
+import argparse
+import os
+
+WINDOWS_LINE_ENDING = b'\r\n'
+UNIX_LINE_ENDING = b'\n'
+
+
+def md5_update_from_dir(directory, hash):
+    assert os.path.isdir(directory)
+    for root, dirnames, filenames in os.walk(directory):
+        for file in sorted(filenames, key=lambda p: str(p).lower()):
+            # If you need include file name to hash uncomment this
+            #hash.update(file.encode())
+            with open(os.path.join(root, file), "rb") as f:
+                hash.update(f.read().replace(WINDOWS_LINE_ENDING, UNIX_LINE_ENDING))
```

Comment:
```suggestion
                hash.update(filter(str.isprintable, f.read()))
```

Reply:
I added replace because git replace line ending for windows

Reply:
why not hash only printable chars? replacing line ending assumes the files contain text, so hashing printable elements is a more robust solution

Reply:
```
 python .\cmd.txt vcpkg
Traceback (most recent call last):
  File ".\cmd.txt", line 32, in <module>
    print(md5_dir(args.folder))
  File ".\cmd.txt", line 25, in md5_dir
    return md5_update_from_dir(directory, hashlib.md5()).hexdigest()
  File ".\cmd.txt", line 18, in md5_update_from_dir
    hash.update(filter(str.isprintable, f.read()))
TypeError: object supporting the buffer API required
```

Reply:
okay lets debug this =)
```suggestion
                for c in filter(str.isprintable, f.read()): hash.update(c)
```

Reply:
> why not hash only printable chars? replacing line ending assumes the files contain text, so hashing printable elements is a more robust solution

The file open in binary mode, if we open as text we also should handle the character encoding. and we will not able to hash binary file.

Reply:
I mean that the contents of the file is a text, not the file open mode. You would not substitute line endings in a binary file like archive or executable.

Reply:
you can try it but it is not trivial to get same hash with python2 python3 windows and linux. maybe your solution is better, but it is not working now.